src/test/javascript/spec/component/exam/manage/exam-exercise-import.component.spec.ts

```diff
@@ -299,4 +299,123 @@ describe('Exam Exercise Import Component', () => {
+
+        programmingExercise2.id = 6;
+        programmingExercise2.title = 'ProgrammingExercise';
+        programmingExercise2.shortName = 'progEx1';
+
+        const programmingExercise3 = new ProgrammingExercise(undefined, exerciseGroup6);
+        programmingExercise3.id = 7;
+        programmingExercise3.title = 'ProgrammingExercise';
+        programmingExercise3.shortName = 'progEx2';
+        exerciseGroup6.exercises = [programmingExercise2, programmingExercise3];
+
+        const exam2 = {
+            id: 10,
+            exerciseGroups: [exerciseGroup1, exerciseGroup2, exerciseGroup3, exerciseGroup4, exerciseGroup5, exerciseGroup6],
+        } as Exam;
+
+        beforeEach(() => {
+            component.exam = exam2;
+            component.importInSameCourse = false;
+            component.ngOnInit();
+        });
+
+        afterEach(() => {
+            programmingExercise2.title = 'ProgrammingExercise';
+            programmingExercise2.shortName = 'progEx1';
+            programmingExercise3.title = 'ProgrammingExercise';
+            programmingExercise3.shortName = 'progEx2';
+        });
+
+        it('should remove duplicated titles when importing', () => {
+            expect(programmingExercise.title).toBe('ProgrammingExercise');
+            expect(programmingExercise2.title).toBe('');
+            expect(programmingExercise3.title).toBe('');
+        });
+
+        it.each([true, false])('should check for programming exercise duplicated titles and short names when entering input', (duplicatedTitles) => {
+            const duplicatesToCheck = duplicatedTitles ? component.exercisesWithDuplicatedTitles : component.exercisesWithDuplicatedShortNames;
+            if (duplicatedTitles) {
+                programmingExercise2.title = programmingExercise.title;
+                programmingExercise3.title = programmingExercise.title;
+            } else {
+                programmingExercise2.shortName = programmingExercise.shortName;
+                programmingExercise3.shortName = programmingExercise.shortName;
+            }
+
+            component.checkForDuplicatedTitlesOrShortNames(programmingExercise, exerciseGroup3, duplicatedTitles);
+
+            expect(duplicatesToCheck.size).toBe(3);
+            expect(duplicatesToCheck.has(programmingExercise.id!)).toBeTrue();
+            expect(duplicatesToCheck.has(programmingExercise2.id!)).toBeTrue();
+            expect(duplicatesToCheck.has(programmingExercise3.id!)).toBeTrue();
+        });
+
+        it.each([
+            [false, false],
+            [false, true],
+            [true, false],
+            [true, true],
+        ])('should remove exercise from duplicates when title / short name is changed', (duplicatedTitles, threeDuplicated) => {
+            const duplicatesToCheck = duplicatedTitles ? component.exercisesWithDuplicatedTitles : component.exercisesWithDuplicatedShortNames;
```

Comment:
Same as above - I find the booleans to be a bit confusing

Reply:
Also for the tests below

Reply:
I changed the first boolean to string and use it as you suggested in another comment. I would keep the second boolean since I am not sure anything else would make it less confusing. You can always look at the name of the argument to see what it does (I renamed it)